tests/integration/suite/test_workloads.py

```diff
@@ -372,3 +415,14 @@ def wait_for_service_cluserip_reset(client, name, timeout=30):
         if time.time() - start > timeout:
             raise Exception('Timeout waiting for workload service')
     return services.data[0]
+
+
+def wait_for_workload(client, name, timeout=30):
+    start = time.time()
+    curr = client.list_workload()["data"][0]
```

Comment:
Does that not error out right away with an index error ?  Feel like we should get a length count first to ensure its above 0. The line below would error out too. 
 https://github.com/rancher/rancher/blob/master/tests/integration/suite/test_app.py#L886-L897

Reply:
I'll fix that, good catch.